src/models/FilterGrower.js

```diff
@@ -57,40 +61,8 @@ export default class Filter {
    * A fn to count the number of current applied filters
    */
   countAppliedFilters() {
-    let numFilters = 0;
-
-    if (this.id) {
-      numFilters += 1;
-    }
-
-    if (this.personId) {
-      numFilters += 1;
-    }
-
-    if (this.deviceIdentifier) {
-      numFilters += 1;
-    }
-
-    if (this.firstName) {
-      numFilters += 1;
+    if (this.organizationId) {
+      return Object.keys(this.getWhereObj()).length;
```

Comment:
Note that we need to account for one case where a non-null/undefined (default) value doesn't contribute towards the filter count:
```
this.organizationId === ALL_ORGANIZATIONS
```
See the original implementation:
```
if (this.organizationId && this.organizationId !== ALL_ORGANIZATIONS) {
  numFilters += 1;
}
```

Reply:
I've made it some tests and when I don't put the if statement, I receive an undefined value for `organization_id`,  so the `Object.keys.length` catches that. In the `getWhereObj` function there isn't a case for `ALL_ORGANIZATIONS`, only if it is different, so when I apply a filter the `organization_id` disappears then it doesn't contribute for the filter count.

[Screencast from 2022-12-12 13-57-29.webm](https://user-images.githubusercontent.com/98126579/207106654-5be4ed86-e0ec-43c4-844e-2bd39636aa49.webm)

Reply:
Thanks @henriquemoreira, that makes sense. Nice work!